packages/catalog-api/src/lib/schema.graphql

```diff
@@ -0,0 +1,237 @@
+transactions that lock on the status field.
+
+Because packages have versions that update over a time, a package may have
+a pending mutation task from initial fetching, or updating by querying and
+fetching new version metadata.
+
+Packages may not exist, in which case we record that they're not found in
+order to avoid flooding npm with invalid requests.
+
+Errors while fetching packages are divided into two categories:
+ - Invalid packages for which we were able to fetch metadata, but something
+in it was incorrect. Hopefully this never happens, but is included for
+completeness.
+ - Errors, such as network failures. In these cases we should be able to
+retry fetching package metadata.
+
+TODO: we need time and possibly version associtated with the last impmort
+attempt.
+"""
+enum PackageStatus {
+  """
+  The package is being downloaded and indexed for the first time.
+  """
+  INITIALIZING
+
+  """
+  Packages may be read from, but there is a pending update task and
+  new versions of the package are being downloaded and indexed.
+  """
+  UPDATING
+
+  """
+  The package is indexed and read to be read.
+  """
+  READY
+
+  """
+  The package was not found on npm.
+  """
+  NOT_FOUND
+
+  """
+  The package metadata was invalid. This is not a recoverable error unless a
+  new package version is published.
+  """
+  INVALID
+
+  """
+  An other recoverable error occured, such as network failure.
+  """
+  ERROR
+}
+
+"""
+Information about a specific version of a package
+"""
+type PackageVersion {
+  version: String!
+
+  """
+  The dist tags assigned to this version
+  """
+  distTags: [String!]!
+
+  """
+  The time this version was written to the registry
+  """
+  lastUpdate: Date!
+
+  status: VersionStatus!
+
+  description: String!
+
+  """
+  The npm package type: 'module' or 'commonjs'.
+  """
+  type: String!
+
+  author: String!
+
+  """
+  The package publication time reported by npm
+  """
+  time: Date!
+
+  homepage: String
+
+  customElements(tagName: String): [CustomElement!]
+
+  customElementsManifest: String
+
+  problems: [ValidationProblem]
+}
+
+enum VersionStatus {
+  """
+  The version is being downloaded and indexed.
+  """
+  INITIALIZING
+
+  """
+  The version is indexed and read to be read.
+  """
+  READY
+
+  """
+  A recoverable error.
+  """
+  ERROR
+}
+
+type ValidationProblem {
+  code: String!
+  severity: String!
+  message: String!
+  filePath: String!
+  start: Int!
+  length: Int!
+}
+
+"""
+Holds important per-element data extracted from the exports of the
+package's custom elements manifest.
+"""
+type CustomElement {
+
+  """
+  The package name containing this element
+  """
+  package: String!
+
+  """
+  The package version containing this element
+  """
+  version: String!
+
+  """
+  The dist tags assigned to this version
+  """
+  distTags: [String!]!
+
+  author: String!
+
+  tagName: String
+
+  className: String
+
+  """
+  Reference to the custom element export
```

Comment:
What's the format for a reference?

Reply:
I need to document it here. It's `{package-name}/{module-path}#{export-name}` where export name is a JS identifier for JS exports and a custom element name for custom element exports.

The code is here: https://github.com/webcomponents/webcomponents.org/blob/main/packages/custom-elements-manifest-tools/src/lib/reference-string.ts